test/e2e/snaps/test-snap-installed-metric.spec.js

```diff
@@ -0,0 +1,107 @@
+const { strict: assert } = require('assert');
+const {
+  defaultGanacheOptions,
+  withFixtures,
+  unlockWallet,
+  getEventPayloads,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
+
+async function mockSegment(mockServer) {
+  return [
+    await mockServer
+      .forPost('https://api.segment.io/v1/batch')
+      .withJsonBodyIncluding({
+        batch: [{ type: 'track', event: 'Snap Installed' }],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+}
+
+describe('Snap Installed Event', function () {
+  it('successfully tracked when snap is installed', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        defaultGanacheOptions,
+        title: this.test.title,
+        testSpecificMock: mockSegment,
+      },
+      async ({ driver, mockedEndpoint: mockedEndpoints }) => {
+        await driver.navigate();
+
+        await unlockWallet(driver);
+
+        // navigate to test snaps page and connect
+        await driver.openNewPage(TEST_SNAPS_WEBSITE_URL);
+        await driver.delay(1000);
+        const confirmButton = await driver.findElement('#connectDialogSnap');
+        await driver.scrollToElement(confirmButton);
+        await driver.delay(500);
+        await driver.clickElement('#connectDialogSnap');
+        await driver.delay(500);
```

Comment:
Why do we need all these delays?

Reply:
Seems to be an issue with the snaps test dapp webpage, removing them prevented the test from progressing. 
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/51488/workflows/3742bccf-bab7-4367-84a3-ba6995d740b5/jobs/1496942
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/51488/workflows/3742bccf-bab7-4367-84a3-ba6995d740b5/jobs/1496943

Seems it is a known issue since all the other snaps test is written this way.
https://github.com/MetaMask/metamask-extension/blob/1277c6225d345188db81386c93e20eee6e588a90/test/e2e/snaps/test-snap-rpc.spec.js#L30-L39

https://github.com/MetaMask/metamask-extension/blob/1277c6225d345188db81386c93e20eee6e588a90/test/e2e/snaps/test-snap-networkaccess.spec.js#L31-L40

https://github.com/MetaMask/metamask-extension/blob/1277c6225d345188db81386c93e20eee6e588a90/test/e2e/snaps/test-snap-networkaccess.spec.js#L31-L40

etc